26-dracula-ant/technical_specification.md

```diff
@@ -22,18 +22,20 @@ Used by the web frontend to create work packages:
   - request body:
     - `dataset_id`: str (the ID of a dataset)
     - `type`: enum (download/upload)
-    - `file_ids`: array of strings  (empty = all files of the dataset)
-    - `public_key`: string (the user's public Crypt4GH key)
+    - `file_ids`: array of strings  (null = all files of the dataset)
+    - `user_public_crypt4gh_key`: string (the user's public Crypt4GH key)
   - response body:
-    - `id`: str (the work package ID)
-    - `token`: str (encrypted work package access token)
+    - `id`: str (the ID of the created work package)
+    - `token`: str (encrypted work and base64 encoded package access token)
+
+*TODO:* The response body could also contain info about the expiration of the token.
```

Comment:
That's a good idea, for now we can just use the lifetime of the work package. We don't need to consider the expiration of the controlled access grant. However, that might come in the future.